permissions/middleware.go

```diff
@@ -64,13 +70,57 @@ func AuthenticationMiddleware() gin.HandlerFunc {
 				}
 
 				c.Set("ID_TOKEN_CLAIMS", claims)
+
+				mdb := c.MustGet("MDB").(*sql.DB)
+				user, err := getOrCreateUser(mdb, &claims)
+				if err != nil {
+					c.AbortWithError(http.StatusBadRequest, err).SetType(gin.ErrorTypePublic)
+					return
+				}
+
+				c.Set("USER", user)
 			}
 		}
 
 		c.Next()
 	}
 }
 
+func getOrCreateUser(mdb *sql.DB, claims *IDTokenClaims) (*models.User, error) {
+	user, err := fetchUserFromDB(mdb, claims.Sub)
+	if err != nil || user != nil {
+		return user, err
+	}
+
+	user = &models.User{
+		AccountsID: null.StringFrom(claims.Sub),
+		Email:      claims.Email,
+		Name:       null.StringFrom(fmt.Sprintf("% %", claims.GivenName, claims.FamilyName)),
+	}
+
+	tx, err := mdb.Begin()
+	utils.Must(err)
+	// check if not unique on DB - "23505": "unique_violation",
+	errDB := user.Insert(tx)
+	if errDB != nil {
+		utils.Must(tx.Rollback())
+		if !strings.Contains(errDB.Error(), "pq: duplicate key value violates unique constraint \"users_accounts_id_key\"") {
```

Comment:
this is wrong. you should not escape this db error. 
See comment above about 2 step migration

Reply:
i wont check that after line 90 but before line 101 from other thread was not created user